src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -1544,6 +1548,170 @@ GetSequencesFromAttrDef(Oid attrdefOid)
 }
 
 
+/*
+ * GetDependentFunctionsWithRelation returns the dependent functions for the
+ * given relation id.
+ */
+List *
+GetDependentFunctionsWithRelation(Oid relationId)
+{
+	List *attrdefResult = NIL;
+	List *constraintResult = NIL;
+	List *dependentFunctionOids = NIL;
+	ScanKeyData key[2];
+	HeapTuple tup;
+
+	Relation depRel = table_open(DependRelationId, AccessShareLock);
+
+	ScanKeyInit(&key[0],
+				Anum_pg_depend_refclassid,
+				BTEqualStrategyNumber, F_OIDEQ,
+				ObjectIdGetDatum(RelationRelationId));
+	ScanKeyInit(&key[1],
+				Anum_pg_depend_refobjid,
+				BTEqualStrategyNumber, F_OIDEQ,
+				ObjectIdGetDatum(relationId));
+
+	SysScanDesc scan = systable_beginscan(depRel, DependReferenceIndexId, true,
+										  NULL, 2, key);
+
+	while (HeapTupleIsValid(tup = systable_getnext(scan)))
+	{
+		Form_pg_depend deprec = (Form_pg_depend) GETSTRUCT(tup);
+
+		/*
+		 * Add columns with default value
+		 */
+		if (deprec->classid == AttrDefaultRelationId &&
+			deprec->objsubid == 0 &&
+			deprec->refobjsubid != 0 &&
+			deprec->deptype == DEPENDENCY_AUTO)
+		{
+			attrdefResult = lappend_oid(attrdefResult, deprec->objid);
+		}
+
+		/*
+		 * Add columns with constraint check
+		 */
+		if (deprec->classid == ConstraintRelationId &&
```

Comment:
When I check pg_catalog, I see that `"table constraint"`, `"default value"`, `"rule"`, `"index"` can be possible when `refobjsubid != 0 `.

Do we need anything for the latter two?

```SQL
select pg_identify_object_as_address(classid, objid, objsubid), pg_identify_object_as_address(refclassid, refobjid, refobjsubid) from pg_depend where refobjsubid != 0  ORDER BY objid DESC;
```

Reply:
Seems like we could add statistics here?
```SQL

-- create statistics based on a valid function
BEGIN;

	CREATE OR REPLACE FUNCTION non_sense_func_for_index()
	RETURNS int
	LANGUAGE plpgsql IMMUTABLE AS
	$$
	BEGIN
	    return 1;
	END; 
	$$;

	CREATE TABLE t3 (
	    a   int,
	    b int
	);

	-- build ndistinct statistics on the pair of expressions (per-expression
	-- statistics are built automatically)
	CREATE STATISTICS s3 ON (a > non_sense_func_for_index()) FROM t3;

	SELECT create_distributed_table('t3', 'a');
ERROR:  function public.non_sense_func_for_index() does not exist
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
CONTEXT:  while executing command on localhost:9700
```

I'm still thinking about removing `deprec->classid ==` checks but let's look a little more

Reply:
Similarly GENERATED ALWAYS:
```SQL
-- generated always
BEGIN;

	CREATE OR REPLACE FUNCTION non_sense_func_for_generated_always()
	RETURNS int
	LANGUAGE plpgsql IMMUTABLE AS
	$$
	BEGIN
	    return 1;
	END; 
	$$;

	CREATE TABLE people (
	id int,
    height_cm numeric,
    height_in numeric GENERATED ALWAYS AS (height_cm / non_sense_func_for_generated_always()) STORED
);

SELECT create_distributed_table('people', 'id');
DETAIL:  on server onderkalaci@localhost:9701 connectionId: 8
ERROR:  function public.non_sense_func_for_generated_always() does not exist
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
CONTEXT:  while executing command on localhost:9700
```

Reply:
Logic for finding dependent functions has been updated and tests above has been added.